core/src/main/scala/kafka/log/remote/RemoteLogManager.scala

```diff
@@ -236,7 +237,10 @@ class RemoteLogManager(fetchLog: TopicPartition => Option[Log],
     // Partitions logs are available when this callback is invoked.
     // Compact topics and internal topics are filtered here as they are not supported with tiered storage.
     def filterPartitions(partitions: Set[Partition]): Set[Partition] = {
-      partitions.filter(partition => !(Topic.isInternal(partition.topic) || partition.log.exists(log => log.config.compact)))
+      partitions.filterNot(partition => Topic.isInternal(partition.topic) ||
+        partition.log.exists(log => log.config.compact || !log.config.remoteStorageEnable) ||
+        partition.topicPartition.topic().equals(TopicBasedRemoteLogMetadataManagerConfig.REMOTE_LOG_METADATA_TOPIC_NAME)
```

Comment:
I agree. This was mentioned as a comment on Satish's PR to trunk too, and he wanted to do it at a later point because that would need more code changes. I can pull in those changes once those are made.